### PR TITLE
post-company

### LINKED ISSUE #4: POST /companies
create companies controller and associated JPA stuff

### PR DESCRIPTION
**Pull request by pmboerth**
added post and get request handling at companies and companies{id} routes

### CODE REVIEW

File: backend/src/main/java/com/internally/backend/advice/CompanyNotFoundAdvice.java [hunk 1]
```diff
@@ -0,0 +1,17 @@
+package com.internally.backend.advice;
```
Comment (aaronkim218) [inline backend/src/main/java/com/internally/backend/advice/CompanyNotFoundAdvice.java]: mayb advices so its plural and stays consistent with controllers exceptions models etc

File: backend/src/main/java/com/internally/backend/models/Company.java [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+        this.name = name;
+        this.industry = industry;
+        this.website = website;
+    }
+
+    public long getId() {
+        return this.id;
+    }
+
+    public void setId(long id) {
```
Comment (aaronkim218) [inline backend/src/main/java/com/internally/backend/models/Company.java]: dont think we will ever need this. is there any scenario where we would change primary key of a company record?

File: backend/src/main/java/com/internally/backend/models/Company.java [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+        this.id = id;
+    }
+
+    public String getName() {
+        return this.name;
+    }
+
+    public void setName(String name) {
```
Comment (aaronkim218) [inline backend/src/main/java/com/internally/backend/models/Company.java:36]: will company class ever need these setters?

are there any scenarios where we will ever need to update company fields?

File: backend/src/main/java/com/internally/backend/advice/CompanyNotFoundAdvice.java (hunk 1 above)
Comment (pmboerth) [reply backend/src/main/java/com/internally/backend/advice/CompanyNotFoundAdvice.java]: yeah should be plural

File: backend/src/main/java/com/internally/backend/models/Company.java (hunk 2 above)
Comment (pmboerth) [reply backend/src/main/java/com/internally/backend/models/Company.java]: probably not, I can remove

File: backend/src/main/java/com/internally/backend/models/Company.java (hunk 3 above)
Comment (pmboerth) [reply backend/src/main/java/com/internally/backend/models/Company.java:36]: not sure, but I feel like having setters is easy enough and if we ever need them in the future

**Review by aaronkim218 — APPROVED** [review]

**Pull request merged**